cohttp/fuzz/fuzz_header.ml

```diff
@@ -206,7 +206,7 @@ let mem_test () =
       [ headers_gen; header_name_gen ] (fun h k ->
         check_eq
           H.(mem h k)
-          List.(mem_assoc k (H.to_list h))))
+          List.(exists (fun (x, _) -> String.lowercase_ascii x = String.lowercase_ascii k) (H.to_list h))))
```

Comment:
The specification and description of this test do not match the implementation. In this case, I think we should probably go for `same behaviour as List.mem_assoc after lowercasing the keys`
```
(* forall h, k. H.mem h k = List.(mem_assoc (String.lowercase_ascii k) (List.map (fun (k, v) -> (String.lowercase_ascii k, v) (H.to_list h))) *)
```

What do you think?

Reply:
Thanks! Your suggestion is indeed a lot nicer than what I had. I've updated the PR with the new test spec + implementation.